### ISSUE TITLE
Calendar Templates / Suggestions

### ISSUE THREAD

**Issue by newhinton** [nextcloud/calendar#4321]
### Is your feature request related to a problem? Please describe.

Sometimes the user wants to create events that are often repeating, but not a "repetition". Currently each event has to be created manually, with all the settings. (Reminder, Location, Title, etc.) This can be cumbersome.

### Describe the solution you'd like

I found the templating-sytem of the android-app 'Business Calendar 2' very intriguing.

In that app, it is possible to save an event as a template. When the user creates a new event, the title-input shows a dropdown with both title-suggestions (generated from past events) and templates. If a template is selected, the other settings will be prefilled.

### Describe alternatives you've considered

It would eben be beneficial if we only provide a dropdown with title-suggestions

**Comment by szaimen** [comment]
Duplicating events will come with probably 3.5.0. Will this cover your usecase?

**Comment by newhinton** [comment]
No, it's not the same. Similar, but there is a difference. 

Duplicating events requires the user to remember and find the event he or she wants to copy, templates would be more an editor-tool. The user creates an event as usual, but instead of "save", he or she "save[s] as template".

Also there would be a dedicated ui to select a template that when selected, prefills all the values in the editor. 

For duplicating, the user needs an existing event-instance, for templates only the editor.

Another way this could be implemented, which is an in-between duplicating&templates, is a 'history'.
If the user creates an event and types the title, this is automatically used as a search, and existing events are proposed. If the user selects one, all the data from that existing event is then prefilled into the new event. 

To see all of this working i can highly recommend to take a look at Business Calendar 2, it has all the mentioned features. I dont propose to do all of them, but i think they could greatly enhance the web-ui especially for regular users.

**Comment by xeruf** [comment]
yes, my boss also uses Business Calendar and seems pretty happy, if we can get NC Calendar as breezy it would be easy to get everyone over :)

**Comment by SDS1337** [comment]
+1 For this. I really need a Template function. Duplicate helps but it's still not as convenient as a Template.

**Comment by matt-ek** [comment]
+1 
It would be very useful if the Nextcloud Text Template functionality ([nextcloud/text_templates](https://github.com/nextcloud/text_templates)) could also be made available in the description field of Calendar events.

This would allow users to quickly insert predefined text snippets when creating or editing events, e.g. for recurring meeting structures, agenda templates, or standardized notes.

As an alternative (or in addition), making the Smart Picker available for the description field would provide a similarly powerful way to insert dynamic or structured content.

This would improve efficiency and consistency when working with events in the Calendar app.

**Comment by SebastianKrupinski** [comment]
Related https://github.com/nextcloud/calendar/issues/7535
